src/utils/security/core.cljs

```diff
@@ -58,3 +60,22 @@
   and does not contain an rtlo character, which might mean that the url is spoofed"
   [text]
   (not (re-matches rtlo-link-regex text)))
+
+(defn hash-masked-password
+  [masked-password]
+  (-> masked-password
+      safe-unmask-data
+      native-module/sha3
+      mask-data))
+
+(defn masked-data-instance?
+  [value]
+  (instance? MaskedData value))
+
+(schema/=> hash-masked-password
```

Comment:
Lovely to see you are using malli already ❤️ 

@clauxx, may I suggest some improvements for readability?

Function schema errors already say if the error is in the input or output schema, so whatever custom schema we define, the error message doesn't need to hardcode if the it's the argument or return value. This allows us to create reusable schemas either for input/output values.

I think by extracting to a custom var schema, we can make the function schema signature clearer, which is one of the main selling points of spec'ing functions. If we leave the custom schema right in the schema definition as you did, this makes the code less readable.

So here's a suggestion:

```clojure
(def ?masked-password
  [:fn {:error/message "should be an instance of utils.security.core/MaskedData"}
   masked-data-instance?])

(schema/=> hash-masked-password
  [:=>
   [:cat ?masked-password]
   ?masked-password])
```

If schemas grow too much in the namespace, I think it's reasonable to create a namespace in the same directory named `schema.cljs`, but I like as you did since there's only one :)

Reply:
Another reason to prefer this style is that custom schemas can easily grow. For example, especially when using `:fn` schemas, malli won't know how to generate random values conforming to the schema, so we would need a `:schema/gen` generator in the definition, thus making the inlined schema less readable still. So by extracting custom schemas to vars we promote more readable function schemas now and in the future as they evolve.

Reply:
yep, it makes perfect sense! I like this way much better. Thanks!!